swift/Samples/AlertContainer/Sources/AlertContainerViewController.swift

```diff
@@ -0,0 +1,258 @@
+    public var alert: Alert?
+    private lazy var title: UILabel = {
+        let title = UILabel()
+        title.font = UIFont.systemFont(ofSize: 18, weight: .medium)
+        title.textAlignment = .center
+        title.translatesAutoresizingMaskIntoConstraints = false
+        return title
+    }()
+    
+    private lazy var message: UILabel = {
+        let message = UILabel()
+        message.font = UIFont.systemFont(ofSize: 18, weight: .medium)
+        message.textAlignment = .center
+        message.numberOfLines = 0
+        message.lineBreakMode = .byWordWrapping
+        message.translatesAutoresizingMaskIntoConstraints = false
+        return message
+    }()
+    
+    public required init(alert: Alert?) {
+        self.alert = alert
+        super.init(frame: .zero)
+        commonInit()
+    }
+    
+    public convenience override init(frame: CGRect) {
+        self.init(alert: nil)
+        self.frame = frame
+        commonInit()
+    }
+    
+    private func commonInit() {
+        if let alert = alert {
+            title.text = alert.title
+            addSubview(title)
+            
+            message.text = alert.message
+            addSubview(message)
+            
+            let buttonStackView = setupButtons(actions: alert.actions)
+            addSubview(buttonStackView)
+            
+            var constraints: [NSLayoutConstraint] = []
+            
+            constraints.append(title.topAnchor.constraint(equalTo: topAnchor, constant: 10))
+            constraints.append(title.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -10))
+            constraints.append(title.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 10))
+            constraints.append(title.heightAnchor.constraint(equalToConstant: 25))
+            
+            constraints.append(message.topAnchor.constraint(equalTo: title.bottomAnchor, constant: 10))
+            constraints.append(message.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -10))
+            constraints.append(message.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 10))
+            
+            constraints.append(buttonStackView.topAnchor.constraint(equalTo: message.bottomAnchor, constant: 15))
+            constraints.append(buttonStackView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: 0))
+            constraints.append(buttonStackView.trailingAnchor.constraint(equalTo: trailingAnchor, constant: 0))
+            constraints.append(buttonStackView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 0))
+            constraints.append(buttonStackView.heightAnchor.constraint(greaterThanOrEqualToConstant: 50))
+            
+            addConstraints(constraints)
+        }
+    }
+    
+    private func setupButtons(actions: [AlertAction]) -> UIStackView {
+        let buttonStackView = UIStackView()
+        buttonStackView.axis = actions.count == 2 ? .horizontal : .vertical
```

Comment:
what happens if `> 2` actions are provided?

Reply:
If we get more than 2 actions, buttons will be stacked vertically similar to how UIAlertController behaves. The only time we make sure buttons are side by side is when there are two buttons. I have not added any functionality to override that.

Reply:
oh ok that makes sense 👍